System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -415,6 +415,11 @@ public override void Move(string sourceDirName, string destDirName)
                 throw new DirectoryNotFoundException($"Could not find a part of the path '{sourceDirName}'.");
             }
 
+            if (!Win32FileSystemBehavior.MoveFileCanParsePath(fullDestPath))
```

Comment:
Did you compare behavior on Windows with behavior on Unix? Do we want to have the same behavior for both?

Reply:
I was thinking about this.  No I haven't manually tested against Unix, though I'm happy to before merging.
Really it seems the best approach for these kind of situations where we're attempting to exactly match the behavior of not just the OS, but the runtime, we could think about a test suite that runs these tests side by side with actual `FileSystem` to ensure the behavior is accurate on the runtime/OS combination.
This does mean introducing I/O into the unit test, but they should still run very fast.
Thoughts?

Reply:
IMO checking it manually is fine. We do not aim for 100% perfect parity with the real filesystem implementation, so I don't think we need to add such tests within this PR.

Reply:
I just manually ran the tests (`dotnet test`) against the real filesystem on both Windows, and Ubuntu (Windows Subsystem for Linux), and they pass.  (Except for the net461 tests on Linux, as expected, as it's unsupported there)
So we are good to go.

Reply:
also added a few more cases to catch some of the stranger pathing possible